context-free.md

```diff
@@ -2,15 +2,19 @@
 
 ## What is easy to do in this language? Why is it easy?
 
+Context Free is a Domain-Specific Language whose limited-expressiveness in terms of its methods and parameters makes it ideal for generating and creating complex patterns and images. It contains parameters such as chirality, transformation, rotation, flipping which create domain focus on image creation since these concepts are all easily-applicable and mostly-specific to images. At the same time, these are concepts comprehensible by most regardless of their experience in graphic design thus increasing the usability of the language to the general public and anybody willing and able to learn it. Its limited-expressiveness in terms of the fact that it defines images only through certain primitive shapes and a few shape and colour readjustment methods means that it can easily be learnt and used by beginners. Generating detailed and patterned images such as fractals is also much easier and more convenient to do using this language as compared to using a typical photo editor.
 
 ## What is hard to do in this language? Why is it hard?
 
+One difficulty in using ContextFree is that its limitation to only using squares, triangles, and circles means that images such as sketches of people, animals, etc. - anything requiring the usage of irregular shapes - are near impossible to produce with accuracy: its usage remains limited to geometric shapes and patterns. In addition to this, when using ContextFree for image creation instead of a photo-editor or a pen and paper, one cannot see the results of their actions and methods (e.g. skewing, flipping, rotating, etc.) in real time and must wait until they have "rendered" the program to do so. Since the designs being created are often complex, this makes it difficult, especially for beginners, to be able to predict the output of a particular program. Were the image able to reflect the changes in the program in real-time, usage of ContextFree may have been much more convenient. Another thing I found difficult to do in this language was debugging. While the program is able to point out the line where the error lies, there are no specifics provided as to the nature of the error thus making it difficult, especially for beginners, to fix their programs.
 
 ## How did you learn how to program in this language?
 
+In order to learn how to program in ContextFree, I used the GitHub documentation for the same. Each part of the syntax - from the startshape to defining rules to shape adjustments to loops and conditionals - were well-defined over there along with some formidable examples. Once I started working on my project, experimenting with the code and changing certain values to compare their expected with their actual effect on the image also helped me learn better about the parameters themselves.
```

Comment:
Same! I also spent a bunch of time on GitHub documentation. And testing also helped with the learning.